bsp/src/org/jetbrains/bsp/protocol/session/BspSession.scala

```diff
@@ -56,6 +56,8 @@ class BspSession private(bspIn: InputStream,
   private val queueProcessor = AppExecutorUtil.getAppScheduledExecutorService
       .scheduleWithFixedDelay(() => nextQueuedCommand, queuePause.toMillis, queuePause.toMillis, TimeUnit.MILLISECONDS)
 
+  def stopServer = serverConnection.cancelable.cancel()
```

Comment:
should be done via org.jetbrains.bsp.protocol.BspCommunicationService#closeCommunication